backend/database/database.go

```diff
@@ -551,11 +591,35 @@ func (d *Database) FavoriteCreate(favorite *Favorite) error {
 
 func (d *Database) Favorite(id uuid.UUID) (*Favorite, error) {
 	const command = `
-		SELECT * FROM Favorite WHERE ID=?
+		SELECT * FROM Favorite WHERE VendorID=?
 	`
 	row := d.db.QueryRowx(command, &id)
 
 	favorite := &Favorite{}
 	err := row.StructScan(favorite)
 	return favorite, err
+
+}
+func (d *Database) FavoritebyVendor(favoriteID uuid.UUID) ([]Favorite, error) {
+	rows, err := d.db.Query("SELECT * FROM Favorite WHERE ID = ?", favoriteID)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	//to hold data that favorite a vendor
+	var fav []Favorite
+
+	//go through rows to assign column to struct field
+	for rows.Next() {
+		var favs Favorite
+		if err := rows.Scan(&favs.ID, &favs.UserID, &favs.VendorID, &favs.DatePosted); err != nil {
```

Comment:
The scan destinations must be in order as defined in `CREATE TABLE`. You can and should use `StructScan` to avoid having to pass individual fields. `StructScan` is only defined for `sqlx.Rows` which is why the other queries are called using [`Queryx()`](https://pkg.go.dev/github.com/jmoiron/sqlx?utm_source=godoc#DB.Queryx).